tests/integration/test_checkpoint_action.py

```diff
@@ -0,0 +1,214 @@
+                "column": "string",
+                "mostly": 1,
+            },
+        )
+    )
+    _ = context.add_or_update_expectation_suite(expectation_suite=expectation_suite)
+    expectation_suite = context.get_expectation_suite(expectation_suite_name=expectation_suite_name)
+    assert (
+        len(expectation_suite.expectations) == 1
+    ), "Expectation Suite was not updated in the previous method call."
+    yield expectation_suite
+    context.delete_expectation_suite(expectation_suite_name=expectation_suite_name)
+    with pytest.raises(get_missing_expectation_suite_error_type):
+        context.get_expectation_suite(expectation_suite_name=expectation_suite_name)
+
+
+@pytest.fixture(scope="module")
+def checkpoint(
+    context: CloudDataContext,
+    data_asset: DataFrameAsset,
+    batch_request: BatchRequest,
+    expectation_suite: ExpectationSuite,
+    get_missing_checkpoint_error_type: type[Exception],
+) -> Iterator[Checkpoint]:
+    checkpoint_name = f"{data_asset.datasource.name} | {data_asset.name}"
+    _ = context.add_checkpoint(
+        name=checkpoint_name,
+        validations=[
+            {
+                "expectation_suite_name": expectation_suite.expectation_suite_name,
+                "batch_request": batch_request,
+            },
+            {
+                "expectation_suite_name": expectation_suite.expectation_suite_name,
+                "batch_request": batch_request,
+            },
+        ],
+    )
+    _ = context.add_or_update_checkpoint(
+        name=checkpoint_name,
+        validations=[
+            {
+                "expectation_suite_name": expectation_suite.expectation_suite_name,
+                "batch_request": batch_request,
+            }
+        ],
+    )
+    checkpoint = context.get_checkpoint(name=checkpoint_name)
+    assert (
+        len(checkpoint.validations) == 1
+    ), "Checkpoint was not updated in the previous method call."
+    yield checkpoint
+    # PP-691: this is a bug
+    # you should only have to pass name
+    context.delete_checkpoint(
+        # name=checkpoint_name,
+        id=checkpoint.ge_cloud_id,
+    )
+    with pytest.raises(get_missing_checkpoint_error_type):
+        context.get_checkpoint(name=checkpoint_name)
+
+
+@pytest.fixture
+def checkpoint_event(checkpoint, datasource_names_to_asset_names):
+    return RunCheckpointEvent(
+        type="run_checkpoint_request",
+        checkpoint_id=checkpoint.ge_cloud_id,
+        datasource_names_to_asset_names=datasource_names_to_asset_names,
+    )
+
+
+def test_loading_context(context):
+    assert isinstance(context, CloudDataContext)
+
+
+def test_running_checkpoint_action(context, checkpoint_event):
+    action = RunCheckpointAction(context=context)
+    event_id = "096ce840-7aa8-45d1-9e64-2833948f4ae8"
+    action_result = action.run(event=checkpoint_event, id=event_id)
+    assert action_result.type == checkpoint_event.type
+    assert action_result.id == event_id
```

Comment:
Love how simple this test is.

Reply:
I know this is a POC but I'm curious if you are planning to retrieve results in cloud (eg the validation result here) and validating what's retrieved is what's expected?